File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& a, int l) {
        sort(a.begin(),a.end());

        int i=0,j=a.size()-1,c=0;

        while(i<=j){
            if(a[i]+a[j]<=l){
                c++;
                i++;
                j--;
            }else if(a[j]<=l){
                c++;
                j--;
            }else{
                c++;
                i++;
            }
        }

        return c;
    }
};

File: 917-boats-to-save-people/boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
         Arrays.sort(people);
        int i = 0, j = people.length - 1;
        int c= 0;

        while (i <= j) {
            c++;
            if (people[i] + people[j] <= limit)
                i++;
            j--;
        }

        return c;
    }
}